lib/connection/pool.js

```diff
@@ -210,6 +210,7 @@ function stateTransition(self, newState) {
   // Get current state
   var legalStates = legalTransitions[self.state];
   if (legalStates && legalStates.indexOf(newState) !== -1) {
+    self.emit('stateChanged', self.state, newState);
```

Comment:
Is this a bug that needs to be fixed?

Reply:
I was actually using this to determine when the pool was destroyed, before I opted to use connection spies. It's a harmless addition, but adds flexibility to the design (it's a no-op if nobody is listening) so I'd opt for keeping it in.

Reply:
Lets make sure that when we squash, this is a separate commit so we can track it.